jni/lib/src/nio/jbyte_buffer.dart

```diff
@@ -0,0 +1,267 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:ffi';
+import 'dart:typed_data';
+
+import '../accessors.dart';
+import '../jarray.dart';
+import '../jni.dart';
+import '../jprimitives.dart';
+import '../jreference.dart';
+import '../jvalues.dart';
+import '../third_party/generated_bindings.dart';
+import '../types.dart';
+import 'jbuffer.dart';
+
+final class JByteBufferType extends JObjType<JByteBuffer> {
+  const JByteBufferType();
+
+  @override
+  String get signature => r"Ljava/nio/ByteBuffer;";
+
+  @override
+  JByteBuffer fromRef(JObjectPtr ref) => JByteBuffer.fromRef(ref);
+
+  @override
+  JObjType get superType => const JBufferType();
+
+  @override
+  final superCount = 2;
+
+  @override
+  int get hashCode => (JByteBufferType).hashCode;
+
+  @override
+  bool operator ==(Object other) {
+    return other.runtimeType == (JByteBufferType) && other is JByteBufferType;
+  }
+}
+
+class JByteBuffer extends JBuffer {
```

Comment:
This could benefit from a long doc comment:

* What it is (single sentence)
* Why to use it: fast memory copying between Java and Dart with `.setRange`.
* Example use snippets
   * How to use it with `asUint8List` with releaseOriginal true
   * How to use it with `asUint8List` with releaseOriginal false